File: backend/app/schemas/conversation.py
"""Conversation and Message schemas."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class ConversationCreateRequest(BaseModel):
    """Create a new conversation."""

    title: str = Field(default="", min_length=0, max_length=255)


class ConversationUpdateRequest(BaseModel):
    """Update conversation title."""

    title: str = Field(min_length=1, max_length=255)


class ConversationListItem(BaseModel):
    """Conversation summary for list view."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    title: str
    created_at: datetime
    updated_at: datetime


class PaginatedConversationList(BaseModel):
    """Paginated list of conversations."""

    items: list[ConversationListItem]
    page: int
    page_size: int
    has_more: bool


class MessageResponse(BaseModel):
    """A single message in a conversation."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    role: str
    content: str
    sources: dict | list | None = None
    prompt_tokens: int | None = None
    completion_tokens: int | None = None
    created_at: datetime


class ConversationDetail(BaseModel):
    """Full conversation with messages."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    title: str
    messages: list[MessageResponse]
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/conversation.ts
export interface ConversationListItem {
  id: string;
  title: string;
  created_at: string;
  updated_at: string;
}

export interface PaginatedConversationList {
  items: ConversationListItem[];
  page: number;
  page_size: number;
  has_more: boolean;
}

import type { Source } from "./chat";

export interface MessageResponse {
  id: string;
  role: "user" | "assistant";
  content: string;
  sources: Source[] | null;
  prompt_tokens: number | null;
  completion_tokens: number | null;
  created_at: string;
}

export interface ConversationDetail {
  id: string;
  title: string;
  messages: MessageResponse[];
  created_at: string;
  updated_at: string;
}
